pkg/runtime/executor.go

```diff
@@ -139,3 +168,68 @@ func (e *Executor) Execute() (results Output, err error) {
 	// successful
 	return results, nil
 }
+
+// read the config file and update scan and skip rules
+func (e *Executor) initScanAndSkipRules() error {
+	if e.configFile != "" {
+		configData, err := config.LoadConfig(e.configFile)
+		if err != nil {
+			zap.S().Error("error loading config file", zap.Error(err))
+			return err
+		}
+
+		if configData.Has(rulesKey) {
+
+			data := (configData.Get(rulesKey)).(*toml.Tree)
+
+			// read scan rules in the toml tree
+			if err := initRules(e, data, scanRulesKey); err != nil {
+				return err
+			}
+
+			// read skip rules in the toml tree
+			if err := initRules(e, data, skipRulesKey); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func initRules(e *Executor, tree *toml.Tree, key string) error {
+	rules, err := getRulesInTomlTree(tree, e.configFile, key)
+	if err != nil {
+		zap.S().Error("error reading config file", zap.Error(err))
+		return err
+	}
+	if len(rules) > 0 {
+		if key == scanRulesKey {
+			e.scanRules = append(e.scanRules, rules...)
```

Comment:
Not the right place to append to scan rules, this function should just return the list of rules. Caller function should take responsibility of using the list of rules. In this way this function can become more generic and we would not require to pass the `Executor` object to this.

Reply:
removed in the refactored code